File: src/main/java/burrows/apps/example/gif/presentation/SchedulerProvider.java
package burrows.apps.example.gif.presentation;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
public final class SchedulerProvider implements BaseSchedulerProvider {
  @Override public Scheduler io() {
    return Schedulers.io();
  }

  @Override public Scheduler ui() {
    return AndroidSchedulers.mainThread();
  }
}

File: src/main/kotlin/burrows/apps/example/gif/presentation/SchedulerProvider.kt
package burrows.apps.example.gif.presentation

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
class SchedulerProvider @Inject constructor() : BaseSchedulerProvider {
  override fun io(): Scheduler = Schedulers.io()
  override fun ui(): Scheduler = AndroidSchedulers.mainThread()
}
